src/common/KokkosKernels_IOUtils.hpp

```diff
@@ -180,26 +180,45 @@ void kk_diagonally_dominant_sparseMatrix_generate(
   {
     int varianz = (1.0*rand()/RAND_MAX-0.5)*row_size_variance;
     rowPtr[row+1] = rowPtr[row] + elements_per_row+varianz;
+    if(rowPtr[row+1] <= rowPtr[row])   // This makes sure that there is
+      rowPtr[row+1] = rowPtr[row] + 1; // at least one nonzero in the row
   }
   nnz = rowPtr[nrows];
   values = new ScalarType[nnz];
   colInd = new OrdinalType[nnz];
   const ScalarType temp = 10;
-  for(OrdinalType row=0;row<nrows;row++)
+  for(OrdinalType row=0; row<nrows; row++)
   {
     ScalarType total_values = 0;
-    for(SizeType k=rowPtr[row] ;k<rowPtr[row+1] - 1;k++)
+    for(SizeType k=rowPtr[row]; k<rowPtr[row+1]-1; k++)
     {
-      OrdinalType pos = row;
-      while (pos == row){
-        pos = ((1.0*rand())/RAND_MAX-0.5)*bandwidth+row;
-      }
-      if(pos<0) pos+=ncols;
+      while (true){
+        OrdinalType pos = (1.0*rand()/RAND_MAX-0.5)*bandwidth+row;
+        while(pos < 0)
+	  pos += ncols;
+        while(pos >= ncols)
```

Comment:
In a later PR probably should add a check that `ncols > 0` in case someone calls the routine with `ncols == 0`

Reply:
else the line below results in inf loop